component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticator.java

```diff
@@ -693,8 +781,13 @@ private boolean isValidTokenLocalUser(int token, String username, Authentication
                         .getUserStoreManager().getUserClaimValues
                                 (tenantAwareUsername, new String[]
                                         {TOTPAuthenticatorConstants.SECRET_KEY_CLAIM_URL}, null);
-                String secretKey = TOTPUtil.decrypt(
-                        userClaimValues.get(TOTPAuthenticatorConstants.SECRET_KEY_CLAIM_URL));
+                String secretKeyClaim = userClaimValues.get(TOTPAuthenticatorConstants.SECRET_KEY_CLAIM_URL);
```

Comment:
Since `secretKeyClaim` will hold the value of the claim we can name it as `secretKeyClaimValue` or something similar, not to be confused with the claim URI.

Reply:
Fixed with bb28af80c0b0082455c678f3e6679107b956d781